trestle/core/control_reader.py

```diff
@@ -799,48 +724,112 @@ def read_implemented_requirement(
-                        )
-                    alter = prof.Alter(
-                        control_id=control_id,
-                        adds=[prof.Add(parts=[part], position='after', by_id=f'{control_id}_smt')]
-                    )
-                    new_alters.append(alter)
-                    found_sections.append(part.name)
+
+        editable_node = None
+        for header in list(control_tree.get_all_headers_for_level(1)):
+            if header.startswith('# Editable'):
+                editable_node = control_tree.get_node_for_key(header)
+                break
+        if not editable_node:
+            return sort_id, [], {}
+
+        after_parts = []
+        ending_parts = {}
+        for subnode in editable_node.subnodes:
+            match = re.match(const.CONTROL_REGEX, subnode.key)
+            if match:
+                part_name_raw = match.groups(0)[0]
+                prose = ControlReader._clean_prose(subnode.content.text)
+                prose = '\n'.join(prose)
+                part_name_snake = spaces_and_caps_to_snake(part_name_raw)
+                part_name = snake_dict.get(part_name_snake, part_name_snake)
+                # if section is required and it hasn't been edited with prose raise error
+                if part_name in required_sections_list and prose.startswith(
+                        const.PROFILE_ADD_REQUIRED_SECTION_FOR_CONTROL_TEXT):
+                    missing_section = sections_dict.get(part_name, part_name)
+                    raise TrestleError(f'Control {control_id} is missing prose for required section {missing_section}')
+                id_ = f'{control_id}_{part_name}'
+                # use sections dict to find correct title otherwise use the title from the markdown
+                part_title = sections_dict.get(part_name, part_name_raw)
+                part = common.Part(id=id_, name=part_name, prose=prose, title=part_title)
+                after_parts.append(part)
+                found_sections.append(part_name)
             else:
-                ii += 1
+                match = re.match(const.PART_REGEX, subnode.key)
+                if not match:
+                    raise TrestleError(f'Unexpected editable header {subnode.key} found in control {control_id}')
+                by_part_label = match.groups(0)[0]
+                control_label_map = label_map.get(control_id, None)
+                if control_label_map is None:
+                    raise TrestleError(f'No label map found for control {control_id}')
+                by_part_id = control_label_map.get(by_part_label, None)
+                if by_part_id is None:
+                    raise TrestleError(f'No part id found for label {by_part_label} in control {control_id}')
+                for node2 in as_list(subnode.subnodes):
+                    if node2.key.startswith('### '):
+                        part_name = node2.key[4:].strip()
```

Comment:
Maybe it is easier to delete hashes rather than hardcoding indices here. Or at least put "### " into the constant so those two are related

Reply:
I now have constant string and remove it using replace with count of 1.